Dockerfile

```diff
@@ -2,23 +2,30 @@ FROM python:3.11-slim AS build
 
 WORKDIR /home/app
 
-COPY requirements/production/requirements.txt .
+RUN pip install poetry
 
-RUN apt-get update && \
-    apt-get install -y --no-install-recommends \
+RUN apt-get update
+RUN apt-get install -y --no-install-recommends \
         build-essential \
         gcc \
         python3-dev \
-        libpq-dev && \
-    pg_config --version && \
-    python -m venv /home/app/venv && \
-    venv/bin/pip install --no-cache-dir -r requirements.txt
+        libpq-dev
+RUN pg_config --version
+
+# poetry will automatically create a ./.venv/
+RUN poetry config virtualenvs.in-project true
 
+COPY pyproject.toml ./
+COPY poetry.lock ./
+RUN poetry install --only main --no-ansi
```

Comment:
Separating these commands into multiple RUN steps results in more layers and a slower build time